src/main/java/dev/blaauwendraad/masker/json/path/JsonPathParser.java

```diff
@@ -0,0 +1,81 @@
+package dev.blaauwendraad.masker.json.path;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parses a jsonpath literal into a {@link dev.blaauwendraad.masker.json.path.JsonPath} object.
+ * <p>
+ * The following features from jsonpath specification are not supported:
+ * <ul>
+ *  <li>Wildcard segments</li>
+ *  <li>Descendant segments</li>
+ *  <li>Wildcard selectors</li>
+ *  <li>Array slice selectors</li>
+ *  <li>Filter selectors</li>
+ *  <li>Function extensions</li>
+ *  <li>Escape characters</li>
+ * </ul>
+ */
+public class JsonPathParser {
```

Comment:
Here is the jsonpath compiler. I decided to name it "Parser" so that it does not evoke a feeling of complexity that is assosiated with "compiler" word.
The parser is under 90 lines of code. It clearly states which jsonpath features are not supported by the library. It is also an extension point in case we need to support those features. 

@gavlyukovskiy @Breus